frontend/src/query_builder/DataReferenceMain.jsx

```diff
@@ -26,6 +26,12 @@ export default class DataReferenceMain extends Component {
             databases = this.props.databases.map((database) => {
                 var dbTables = this.state.databases[database.id];
                 if (dbTables === undefined) {
+                    if (database.tables) {
+                        this.state.databases[database.id] = database.tables;
```

Comment:
@tlrobinson why are we setting state in the render() function on this component?